src/main.c

```diff
@@ -431,9 +431,15 @@ int main(int argc, char **argv)
 		goto user_error;
 	}
 	// If no password given, interactively ask user
-	if (cfg.password[0] == '\0')
-		read_password("VPN account password: ", cfg.password,
-		              FIELD_SIZE);
+	if (cfg.password == NULL || cfg.password[0] == '\0') {
+		if (cfg.password != NULL) {
+			free(cfg.password);
+		}
+		char *tmp_password = malloc(BUFSIZ); // allocate large buffer
```

Comment:
Indeed `BUFSIZ` is the [size of the buffer assigned to a stream by _setbuf_()](http://pubs.opengroup.org/onlinepubs/9699919799/functions/setbuf.html). Why use `BUFSIZ` here?

Reply:
If we allocate more fields dynamically then we don't need something large any more. Perhaps use chunks of 64 bytes as 64 bytes ought to be enough for most users.

Reply:
64 bytes is pretty small, even if we use a buffer on the stack I would advise something bigger like 1k or 4k ; I don't really care about the value, I'd rather avoid depending on an external define we don't control - we actually even redefine a BUFSZ in http.c so it can be confusing...

To be honest though I'd rather avoid an explicit loop, if we want fully dynamic the simplest way is to make read_password use getline: now we still have ICANON in the terminal there is no reason not to let libc handle it, I played with it and this appears to work https://github.com/martinetd/openfortivpn/commit/8c137d9004f5777e8f974c756ef7336b23cc2ece (I left a printf/return just after read_password to test, this isn't meant to take as is)
The problem with this approach is if someone were to do something as simple as `openfortivpn < /dev/zero`, openfortivpn's memory usage would grow unchecked; if you don't have much swap that's fine and if there is swap linux will become unresponsive for a while before openfortivpn finally gets killed by OOM....

Ultimately I think a fixed-but-large-enough buffer size is better, 4k or 8k is fine to have on the stack and read_password could do a single read() call and not worry about partial reads or whatsnot - just trusting the terminal or complaining if newline wasn't found - I just don't like external macros :)